Bamboo/backend/bamboo-backend/src/routes/auth/initialise/initialise.service.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable prettier/prettier */
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { UserService } from 'src/services/user/user.service';
+import { AccountJwtPayload } from 'src/types/auth.account.types';
+import { JwtPayload } from 'src/types/auth.jwt.types';
+
+@Injectable()
+export class InitialiseService {
+    constructor(
+        private usersService: UserService,
+        private jwtService: JwtService,
+    ) {}
+
+    async initialiseAccount(accountType: string, jwt: JwtPayload){
+        const account = await this.usersService.initialiseAccount(
+            accountType,
+            jwt
+        )
+
+        // Generate a new JWT token with the updated account
+        const firstName = account.name.split(' ')[0]
+        const lastName = account.name.split(' ')[1]
```

Comment:
dont use string.split methods when determining firstName and lastName. These should be separate attributes

Reply:
userService.initialiseAccount response type should match the response type of userService.createUser or .findOrCreate